test/conformance/ingress/path_test.go

```diff
@@ -210,27 +213,40 @@ func TestPathAndPercentageSplit(t *testing.T) {
 	defer cancel()
 
 	const (
-		total     = 100
+		total     = 1000
 		totalHalf = total / 2
-		tolerance = total * 0.20
+		tolerance = total * 0.15
 	)
 	got := make(map[string]float64, 2)
 	wantKeys := sets.NewString(fooName, barName)
-	for i := 0; i < total; i++ {
-		ri := RuntimeRequest(t, client, "http://"+name+".example.com/foo")
-		if ri == nil {
-			return
-		}
+	wg := pool.New(total)
 
-		gotH := ri.Request.Headers.Get(headerName)
-		got[gotH]++
+	var mu sync.Mutex
+	for i := 0; i < total; i++ {
+		wg.Go(func() error {
+			ri := RuntimeRequest(t, client, "http://"+name+".example.com/foo")
+			if ri == nil {
+				return fmt.Errorf("failed to request")
+			}
+			gotH := ri.Request.Headers.Get(headerName)
+			mu.Lock()
+			defer mu.Unlock()
+			got[gotH]++
```

Comment:
Would it be simpler to write the header to a channel and count from that channel synchronously? We usually do that but I'm kinda indifferent in this case.

The pattern would be:

```golang
resultChan := make(chan string, total)
for ... {
  wg.Go(func() error {
    ...
    resultCh <- ri.Request.Headers.Get(headerName)
  }
}
wg.Wait()
close(resultChan)
for _, r := range resultChan {
  // do stuff
}
```

Writing it out, your version is probably a lot shorter 😂